command/pr_create.go

```diff
@@ -15,6 +15,39 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type defaults struct {
+	Title string
+	Body  string
+}
+
+func computeDefaults(baseRef, headRef string) (defaults, error) {
+	commits, err := git.Commits(baseRef, headRef)
+	if err != nil {
+		return defaults{}, err
+	}
+
+	out := defaults{}
+
+	if len(commits) == 1 {
+		out.Title = commits[0].Title
+		body, err := git.CommitBody(commits[0].Sha)
+		if err != nil {
+			return defaults{}, err
+		}
+		out.Body = body
+	} else {
+		out.Title = headRef // TODO format or something?
+
+		body := fmt.Sprintf("---\n%d commits:\n\n", len(commits))
+		for _, c := range commits {
+			body += fmt.Sprintf("- %s %s\n", c.Sha[0:5], c.Title)
```

Comment:
I wonder if we could get away with not printing the SHA for the commits at all, just subject lines? I don't consider the SHAs very useful when crafting a pull request message

Reply:
i don't have strong feelings. seeing the SHAs has helped me reality check in the past during a rebase so I included them reflexively but if others don't think there is utility i'm fine taking them out.